src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs

```diff
@@ -43,14 +46,31 @@ public override Task Publish(Message message)
             return Task.WhenAll(messages.Select(m => ProcessMessage(m)));
         }
 
-        private Task ProcessMessage(AppMessage message)
+        private async Task ProcessMessage(AppMessage message)
         {
             if (message is HubMessage hubMessage)
             {
-                return WriteMessage(_serviceConnectionManager.WithHub(hubMessage.HubName), message);
+                try
+                {
+                    await WriteMessage(_serviceConnectionManager.WithHub(hubMessage.HubName), message);
+                    if (hubMessage.Message is IMessageWithTracingId msg && msg.TracingId != null)
```

Comment:
discussion: is moving the check into Log better?

Reply:
I don't think so. Actually I try to hide the check inside Log in RT, but later I need to modify some check in one of the callers, and I am not sure in what condition I log it if I don't view the definition of the log, not convenient if you consider it.

Reply:
Can the message be `IMessageWithTracingId` but its `TracingId` be null?

Reply:
yeah, maybe move the `TracingId` null check into Log? BTW, can message not `IMessageWithTracingId`? it feels annoying to do some if check before Logging everytime

Reply:
that's valid for a message, we only need to log in 2 cases:
* message sent from diagnostic client
* the service enable messaing log
Before messages reaches here, the tracing ID is already generated (in ServiceLifetimeManagerBase). 
So we only need to log the message with non-null tracing ID

Reply:
if the message isn't `IMessageWithTracingId`, I won't log it. For those messages I'm insterested, I already implement the `IMessageWithTracingId`